src/overlays/actors/ovl_En_Ossan/z_en_ossan.c

```diff
@@ -37,227 +114,2406 @@ const ActorInit En_Ossan_InitVars = {
+            this->tempStateFlag = this->stateFlag;
+            func_8010B720(globalCtx, this->shelfSlots[this->cursorIndex]->itemBuyPromptTextId);
+            this->stickRightPrompt.isEnabled = false;
+            this->stickLeftPrompt.isEnabled = false;
+            switch (selectedItem->actor.params) {
+                case SI_KEATON_MASK:
+                case SI_SPOOKY_MASK:
+                case SI_SKULL_MASK:
+                case SI_BUNNY_HOOD:
+                case SI_MASK_OF_TRUTH:
+                case SI_ZORA_MASK:
+                case SI_GORON_MASK:
+                case SI_GERUDO_MASK:
+                    func_80078884(NA_SE_SY_DECIDE);
+                    this->drawCursor = 0;
+                    this->stateFlag = ENOSSAN_STATE_SELECT_ITEM_MASK;
+                    return true;
+                case SI_MILK_BOTTLE:
+                    func_80078884(NA_SE_SY_DECIDE);
+                    this->drawCursor = 0;
+                    this->stateFlag = ENOSSAN_STATE_SELECT_ITEM_MILK_BOTTLE;
+                    return true;
+                case SI_WEIRD_EGG:
+                    func_80078884(NA_SE_SY_DECIDE);
+                    this->drawCursor = 0;
+                    this->stateFlag = ENOSSAN_STATE_SELECT_ITEM_WEIRD_EGG;
+                    return true;
+                case SI_19:
+                case SI_20:
+                    func_80078884(NA_SE_SY_ERROR);
+                    this->drawCursor = 0;
+                    this->stateFlag = ENOSSAN_STATE_SELECT_ITEM_UNIMPLEMENTED;
+                    return true;
+                case SI_BOMBS_5_R25:
+                case SI_BOMBS_10:
+                case SI_BOMBS_20:
+                case SI_BOMBS_30:
+                case SI_BOMBS_5_R35:
+                    func_80078884(NA_SE_SY_DECIDE);
+                    this->drawCursor = 0;
+                    this->stateFlag = ENOSSAN_STATE_SELECT_ITEM_BOMBS;
+                    return true;
+                default:
+                    func_80078884(NA_SE_SY_DECIDE);
+                    this->drawCursor = 0;
+                    this->stateFlag = ENOSSAN_STATE_SELECT_ITEM;
+                    return true;
+            }
+        }
+        func_80078884(NA_SE_SY_ERROR);
+        return true;
+    }
+    return false;
+}
+
+void EnOssan_State_BrowseLeftShelf(EnOssan* this, GlobalContext* globalCtx, Player* player) {
+    s32 a;
+    s32 b;
+    u8 prevIndex = this->cursorIndex;
+    s32 c;
+    s32 d;
+
+    if (!EnOssan_ReturnItemToShelf(this)) {
+        osSyncPrintf("%s[%d]:\x1b[32mズーム中！！\x1b[m\n", "../z_en_oB1.c", 2152);
+        this->delayTimer = 3;
+        return;
+    }
+    if (this->delayTimer != 0) {
+        this->delayTimer--;
+        return;
+    }
+    this->drawCursor = 0xFF;
+    this->stickLeftPrompt.isEnabled = true;
+    EnOssan_UpdateCursorPos(globalCtx, this);
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 5) &&
+        !EnOssan_HasPlayerSelectedItem(globalCtx, this, globalCtx->state.input)) {
```

Comment:
From what I can tell, functions that take inputs here a pointer to a specific Input struct, so I think they should take `&globalCtx->state.input[0]` instead of `globalCtx->state.input` (relevant in some other places in the file)
```suggestion
        !EnOssan_HasPlayerSelectedItem(globalCtx, this, &globalCtx->state.input[0])) {
```